### PR TITLE
[code2fn] Creates model ingestion coverage report generation script 

### LINKED ISSUE #577: Automating Model Coverage Analysis
Come up with a way to automatically analyze and report, for any model
- How many files we can successfully ingest
- How many files we can't currently ingest
- Report these findings in some kind of tabular form (HTML?) and with some kind of percentage coverage of ingested files
 
Extra ideas
- Error messages describing why we can't ingest files
- Line by line reporting as opposed to per file reporting (using tree-sitter?)
- Idiom tracking (what's covered vs what isn't covered) per language

### PR DESCRIPTION
**Pull request by vincentraymond-ua**
## Model Coverage Report
Creates script to automatically generate reports detailing coverage of model ingestion by the code2fn pipeline. For each file, the script will run every step of the code2fn pipeline (tree-sitter, cast, gromet) and produce a corresponding data product. If no error occurs and each data product is non-empty, the script marks the file as ingestible. 

To reduce overhead, there is a 10 second timeout for each step. However, no current model is triggering this timeout.

### CI Integration
Adds report generating step to Github CI. With the current set of models, this step takes ~2.5 minutes and generates ~300MB. This process should also automatically add the report to the [Github Site](https://ml4ai.github.io/skema/)
### Wiki Page
For information on how to add a new model to the project, see : https://github.com/ml4ai/skema/wiki/Adding-a-new-model-to-the-project

For information on running the report generator, see: 
https://github.com/ml4ai/skema/wiki/Generating-Code2fn-Coverage-Report

## Other changes
- Adds .yaml, .yml, and .html to allowed data files being packaged with Skema 
- Adds skema.utils path to pyproject.toml
- Adds utility functions to skema/program_analysis/tree_sitter_parsers to make it easier to map from extension->language_name
- Adds func_timeout dependency to pyprojects.toml

Resolves #577

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ nav:
         - Configuring <code>img2mml</code>: "dev/eqn2mml.md"
         - Building documentation: "dev/docs.md"
         - Test coverage reports: "coverage/index.md"
+        - Code2fn coverage reports: "coverage/code2fn_coverage/report.html"
```
Comment (myedibleenso) [inline mkdocs.yml:42]: Nice!

File: pyproject.toml [hunk 2]
```diff
@@ -101,7 +103,7 @@ all = ["skema[core]", "skema[dev]", "skema[doc]", "skema[demo]", "skema[annotati
 
 [tool.setuptools.package-data]
 # needed to ensure models are included in package/discoverable
-"*" = ["*.json", "vocab.txt", "*.pt", "*.png"]
+"*" = ["*.json", "vocab.txt", "*.pt", "*.png", "*.html", "*.yml", "*.yaml"]
```
Comment (myedibleenso) [inline pyproject.toml:106]: @vincentraymond-ua , do we need to include .html in the package source? Site publishing ignores these settings.

Comment (myedibleenso) [reply pyproject.toml:106]: nm.  I see why this is needed now.

**Review by myedibleenso — APPROVED** [review]
Great work!  What is the issue with the zip archive for climlab v2?

**Comment by vincentraymond-ua** [comment]
> Great work! What is the issue with the zip archive for climlab v2?

#599 - There are MACOS artifacts included in the zip archive that are causing ingestion issues.

**Pull request merged**